src/components/batch-selection/batch-selection.test.tsx

```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import BatchSelection from ".";
+import Button from "../button";
+import ButtonMinor from "../button-minor";
+import Icon from "../icon";
+import IconButton from "../icon-button";
+import Link from "../link";
+
+test("Renders with children", () => {
+  render(
+    <BatchSelection selectedCount={0}>
+      <IconButton>
+        <Icon type="edit" />
+      </IconButton>
+    </BatchSelection>
+  );
+
+  const iconButton = screen.getByRole("button", { name: "edit" });
+  expect(iconButton).toBeVisible();
+  expect(iconButton).toBeEnabled();
+});
+
+test("Renders as hidden when the `hidden` prop is true", () => {
+  render(
+    <BatchSelection selectedCount={0} hidden>
+      <IconButton>
+        <Icon type="bin" />
+      </IconButton>
+    </BatchSelection>
+  );
+
+  const batchSelection = screen.getByTestId("batch-selection");
+  expect(batchSelection).toHaveAttribute("hidden");
+});
+
+test("`IconButton` children should be automatically disabled via context", () => {
+  render(
+    <BatchSelection colorTheme="dark" selectedCount={0} disabled>
+      <IconButton>
+        <Icon type="edit" />
+      </IconButton>
+    </BatchSelection>
+  );
+
+  const iconButton = screen.getByRole("button", { name: "edit" });
+
+  expect(iconButton).toBeDisabled();
+});
+
+test("`Button` children should be automatically disabled via context", () => {
+  render(
+    <BatchSelection colorTheme="light" selectedCount={0} disabled>
+      <Button iconType="edit" />
+    </BatchSelection>
+  );
+
+  const button = screen.getByRole("button", { name: "edit" });
+
+  expect(button).toBeDisabled();
+});
+
+test("`ButtonMinor` children should be automatically disabled via context", () => {
+  render(
+    <BatchSelection colorTheme="white" selectedCount={0} disabled>
+      <ButtonMinor iconType="edit" />
+    </BatchSelection>
+  );
+
+  const minorButton = screen.getByRole("button", { name: "edit" });
+
+  expect(minorButton).toBeDisabled();
+});
+
+test("`Link` children should be automatically disabled via context", () => {
+  render(
+    <BatchSelection selectedCount={0} disabled>
+      <Link>Link as a button</Link>
+    </BatchSelection>
+  );
+
+  const link = screen.getByTestId("link-anchor");
+  expect(link).toHaveStyle("cursor: not-allowed");
+  link.focus();
+  expect(link).not.toHaveFocus();
```

Comment:
nitpick of formatting:

```suggestion
  link.focus();
  
  expect(link).not.toHaveFocus();
```